src/osd/PrimaryLogPG.cc

```diff
@@ -8602,6 +8609,47 @@ void PrimaryLogPG::finish_ctx(OpContext *ctx, int log_op_type, int result)
 	   << dendl;
   utime_t now = ceph_clock_now();
 
+  // Drop the reference if deduped chunk is modified
+  if (ctx->new_obs.oi.is_dirty() &&
+    (ctx->obs->oi.has_manifest() && ctx->obs->oi.manifest.is_chunked()) &&
+    // If a clone is creating, ignore dropping the reference for manifest object
+    !ctx->delta_stats.num_object_clones &&
+    ctx->new_obs.oi.size != 0 && // missing, redirect and delete
+    !ctx->cache_operation &&
+    log_op_type != pg_log_entry_t::PROMOTE) {
+    object_manifest_t updated_chunks;
+    const object_info_t &oi = ctx->obs->oi;
+    updated_chunks.type = object_manifest_t::TYPE_CHUNKED;
+    // try to find out intersections
+    for (auto &p : oi.manifest.chunk_map) {
+      if (is_dedup_chunk(oi, p.second)) {
+	if (!ctx->clean_regions.is_clean_region(p.first, p.second.length) &&
+	    p.second.oid != hobject_t()) {
```

Comment:
This check should never be necessary right?  We're never ever going to leave a chunk_map entry with an empty reference.